### PR TITLE
Change pointer to struct on return from GetDeployment

### PR DESCRIPTION
**Pull request by paulmey**

### CODE REVIEW

File: management/virtualmachine/client.go [hunk 1]
```diff
@@ -38,25 +38,24 @@ func (self VirtualMachineClient) CreateDeployment(role Role, cloudServiceName st
 	return self.client.SendAzurePostRequest(requestURL, data)
 }
 
-func (self VirtualMachineClient) GetDeployment(cloudServiceName, deploymentName string) (*DeploymentResponse, error) {
+func (self VirtualMachineClient) GetDeployment(cloudServiceName, deploymentName string) (DeploymentResponse, error) {
+	deployment := DeploymentResponse{}
```
Comment (ahmetalpbalkan) [inline management/virtualmachine/client.go]: declare with `var` if you're not gonna set any fields. (sorta convention and easier to read)

File: management/virtualmachine/client.go [hunk 2]
```diff
@@ -38,25 +38,24 @@ func (self VirtualMachineClient) CreateDeployment(role Role, cloudServiceName st
...
-
 	requestURL := fmt.Sprintf(azureDeploymentURL, cloudServiceName, deploymentName)
 	response, azureErr := self.client.SendAzureGetRequest(requestURL)
 	if azureErr != nil {
-		return nil, azureErr
+		return deployment, azureErr
 	}
 
 	err := xml.Unmarshal(response, deployment)
 	if err != nil {
```
Comment (ahmetalpbalkan) [inline management/virtualmachine/client.go]: isn't this if obsolote now?

**Comment by paulmey** [comment]
To merge or not to merge?

**Comment by ahmetalpbalkan** [comment]
@paulmey please ping next time you push the fixes. :) otherwise I don't have any way to notice it.

**Comment by paulmey** [comment]
sure, thanks!

**Pull request merged**